inspirehep/dojson/hep/fields/bd1xx.py

```diff
@@ -141,6 +145,30 @@ def _get_record(value):
             if x_value and x_value.isdigit():
                 return get_record_ref(x_value, 'authors')
 
+        def _get_contributor_role(value):
+            if not value:
+                return None
+            if not isinstance(value, str):
+                value = set(value)
+                if len(value) is 1:
+                    value = value.pop()
```

Comment:
I think here you should use one of the utilities `force_force_list()`, I believe we have mentioned IRL that one author can have a list of contributor roles, so you should force your input to be a list using the helper and then happily append all the mapped values.